File: 2. Add Two Numbers/2. Add Two Numbers.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
        head = ListNode(0)
        temp = head
        carry = 0
        #while l1 is not None || l2 is not None || carry != 0:
        while l1 is not None or l2 is not None or carry !=0:
            sum = 0
            if l1 is not None:
                sum = sum+ l1.val
                l1=l1.next
            if l2 is not None:
                sum = sum+ l2.val
                l2=l2.next
            sum = sum+ carry
            temp.next = ListNode(sum%10)
            temp = temp.next
            carry = sum//10
        return head.next

        

File: 2. Add Two Numbers/2. Add Two Numbers.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* addTwoNumbers(ListNode* l1, ListNode* l2) {
        ListNode *ansNode = new ListNode();
        ListNode *returnNode = ansNode;

        int carry = 0;
        int sum = 0;
        while(l1 or l2)
        {
            sum = 0;
            if(l1)
            {
                sum += l1->val;
                l1 = l1->next;
            }
            if(l2)
            {
                sum += l2->val;
                l2 = l2->next;
            }
            sum += carry;
            carry = sum / 10;
            returnNode->next = new ListNode(sum%10);
            returnNode = returnNode->next ;
        }
        if(carry != 0)
        {
            returnNode->next = new ListNode(carry);
        }
        return ansNode->next;
    }
};